### PR TITLE
Some light mysql service enhancements

### PR DESCRIPTION
**Pull request by joachifm**
I think these are fairly self explanatory. The `preStart` permission patch does fix some classes of problems that would otherwise require manual intervention, however.

### CODE REVIEW

**Review by groxxda — APPROVED** [review]
lgtm

File: nixos/modules/services/databases/mysql.nix [hunk 1]
```diff
@@ -180,12 +194,14 @@ in
...
             if ! test -e ${cfg.dataDir}/mysql; then
-                mkdir -m 0700 -p ${cfg.dataDir}
-                chown -R ${cfg.user} ${cfg.dataDir}
+                mkdir -p ${cfg.dataDir}
                 ${mysql}/bin/mysql_install_db ${mysqldOptions}
                 touch /tmp/mysql_init
             fi
 
+            chmod -m 700 ${cfg.dataDir}
+            chown -R ${cfg.user} ${cfg.dataDir}
```
Comment (grahamc) [inline nixos/modules/services/databases/mysql.nix]: :-1: on this, this is seriously expensive on a big deployment.

**Comment by grahamc** [comment]
:+1: on the changes, but :-1: on "fixing" the permissions every run.

**Review by grahamc — CHANGES_REQUESTED** [review]

File: nixos/modules/services/databases/mysql.nix (hunk 1 above)
Comment (joachifm) [inline nixos/modules/services/databases/mysql.nix]: I see. Is there an alternative solution?

Comment (joachifm) [inline nixos/modules/services/databases/mysql.nix]: It's possible to ask chmod/chown to only update permissions if they differ, is that better?

**Comment by joachifm** [comment]
I've dropped the `preStart` change, though I still think allowing changing users without ensuring consistency makes for a poor interface.

**Pull request merged**